metron-interface/metron-config/e2e/sensor-config-readonly/sensor-config-readonly.po.ts

```diff
@@ -0,0 +1,125 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { browser, element, by, protractor } from 'protractor/globals';
+import {changeURL, waitForElementVisibility} from '../utils/e2e_util';
+
+export class SensorDetailsPage {
+
+    private enableButton;
+    private disableButton;
+    private startButton;
+    private stopButton;
+
+    constructor() {
+        this.enableButton = element(by.cssContainingText('metron-config-sensor-parser-readonly .btn', 'ENABLE'));
+        this.disableButton = element(by.cssContainingText('metron-config-sensor-parser-readonly .btn', 'DISABLE'));
+        this.startButton = element(by.cssContainingText('metron-config-sensor-parser-readonly .btn', 'START'));
+        this.stopButton = element(by.cssContainingText('metron-config-sensor-parser-readonly .btn', 'STOP'));
+    }
+
+    clickToggleShowMoreLess(text: string, index: number) {
+        return element.all(by.linkText(text)).get(index).click().then(() => {
+            browser.sleep(1000);
+            return true;
+        })
+    }
+
+    closePane(name: string) {
+        return element(by.css('metron-config-sensor-parser-readonly .fa-times')).click().then(() => {
+            return true;
+        });
+    }
+
+    disableParser() {
+        return waitForElementVisibility(this.disableButton).then(() => {
+            return this.disableButton.click().then(() => {
+                return waitForElementVisibility(this.enableButton).then(() => {
+                    return true;
+                })
+            });
+        });
+    }
+
+    enableParser() {
+        return waitForElementVisibility(this.enableButton).then(() => {
+            return this.enableButton.click().then(() => {
+                return waitForElementVisibility(this.disableButton).then(() => {
+                    return true;
+                })
+            });
+        });
+    }
+
+    startParser() {
+        return waitForElementVisibility(this.startButton).then(() => {
+            return this.startButton.click().then(() => {
+                return waitForElementVisibility(this.stopButton).then(() => {
+                    return true;
+                })
+            });
+        });
+    }
+
+    stopParser() {
+        return waitForElementVisibility(this.stopButton).then(() => {
+            return this.stopButton.click().then(() => {
+                return waitForElementVisibility(this.startButton).then(() => {
+                    return true;
+                })
+            });
+        });
+    }
+
+    getButtons() {
+        return element.all(by.css('metron-config-sensor-parser-readonly button:not([hidden=""])')).getText();
+    }
+
+    getGrokStatement() {
```

Comment:
My only ask is that as part of the architecture document, we get a clear answer to the following question:

I am a parser author who has a parser which requires configuration.  Where and how do I add that configuration screen?

Based on the answer to that question, I think we can figure out if that's going to scale to many sensors, each with their own configs.  If it does not, then I'd ask that we at least have a discuss thread where we can create a migration path.

I'll admit that I'm no UI engineer, but we all have to understand this so we can contribute to it, so I apologize if my questions are naive. :)